### PR TITLE
HG-686 removed namespace specifier from random page query for community wiki

### PR DESCRIPTION
**Pull request by katanka**
https://wikia-inc.atlassian.net/browse/HG-686

### CODE REVIEW

File: server/lib/MediaWiki.ts [hunk 1]
```diff
@@ -153,12 +153,21 @@ export class ArticleRequest {
 	 * @return {Promise<any>}
 	 */
 	randomTitle (): Promise<any> {
-		var url = createUrl(this.wikiDomain, 'api.php', {
-			action: 'query',
-			generator: 'random',
-			grnnamespace: 0,
-			format: 'json'
-		});
+		var url = "";
```
Comment (rogatty) [inline server/lib/MediaWiki.ts]: Use single quotes.

**Comment by rogatty** [comment]
@katanka can you explain this change? How does removing the namespace fix the bug? Won't it return objects that Mercury doesn't support (non-articles)?

**Comment by katanka** [comment]
@rogatty For community.wikia.com, that is the correct functionality. If you try the random page link on the desktop (non-mercury) site, it will go to all of the namespaces. So by removing the namespace specifier for just community, it makes the functionality identical on both platforms.

**Comment by rogatty** [comment]
But will it work? What happens when non-article is returned? I believe that somewhere in the code we assume it will always be an article.

**Comment by kenkouot** [comment]
Would it return the full namespace + title? EG: "Help:Foo"?

File: server/lib/MediaWiki.ts [hunk 2]
```diff
@@ -153,12 +153,21 @@ export class ArticleRequest {
...
+		if (this.wikiDomain.indexOf('community') > -1) {
```
Comment (hakubo) [inline server/lib/MediaWiki.ts]: also can we restrain from doing special cases for specific communities?
maybe we can do 'global' fix? 

We have this pattern in many places and it's hard to reason what happens where after some time.

Comment (katanka) [inline server/lib/MediaWiki.ts]: Yeah, I agree it isn't pretty. However, it sorta is a special case: we need behavior on the Community wiki that is different from the other wikis. Is there a better way to implement it?

**Comment by katanka** [comment]
@kenkouot Yes, it returns [namespace]:[title]

@rogatty As far as I can tell, that bug was caused by the fact that random was only returning pages in the Main namespace, which only has ~60 pages in it. So, it was very likely to get repeats. I was not able to replicate it returning the same page over and over again. 

By opening it up to all the namespaces, like the desktop site does, we only get repeats at a very small rate.

**Comment by rogatty** [comment]
@katanka:
Open http://gameofthrones.wikia.com/api.php?action=query&generator=random&grnnamespace=0&format=json in your browser and refresh multiple times. Each time I do it I get a different article.
Now open http://gameofthrones.wikia.com/ with Mercury and try getting the random article. I clicked ~20 times and got two articles alternately.

File: server/lib/MediaWiki.ts (hunk 1 above)
Comment (rogatty) [inline server/lib/MediaWiki.ts]: tslint warning: `(quotemark) lib/MediaWiki.ts[156, 13]: " should be '`
@katanka please look at the console before doing commits.

**Comment by rogatty** [comment]
@kenkouot why did you merge it? Now on Community after clicking on random page I get comments, user pages and a lot of "Sorry, this article seems to be empty!". Those pages aren't supported by Mercury and we shouldn't even try to load them.

I'm reverting the merge because it introduces regression - broken pages.

Please start doing QA.

/cc @hakubo @hobbs @jkuner

**Comment by rogatty** [comment]
@katanka can you create a proper PR with change that was described in the ticket (fixing cache issue)? It's this one: https://github.com/Wikia/mercury/pull/887/files#diff-826bec53995755be79d0d75d93dd48e3R38

Thanks.

**Pull request merged**